### ISSUE TITLE
starting BoltDB permission denied issue

### ISSUE THREAD

**Issue by sangkyu-korbit** [runatlantis/helm-charts#343]
Hello!

Currently, we have linked Atlantis service using self-managed gitlab.

The implementation was applied according to the official document help chart implementation method (up to terragrunt), and the following error occurs when the help chart is applied.

`Error: initializing server: starting BoltDB: open /atlantis-data/atlantis.db: permission denied`

![Gitlab atlantis plan error](https://github.com/runatlantis/helm-charts/assets/103010807/9316ad27-848a-41a3-ae43-d800c652b218)

This symptom occurs when you set only the orgAllowlist and gitlab part of the default chart, and when the runAsUser is set to 1000, the pod works normally.

However, if the gitlab sends the 'atlantis plan' command when you change it like that, the following error will occur.

```
* * error downloading 'https://gitlab.xxx.com/devops/xxx-terraform-module.git?ref=v1.10.7': /usr/bin/git exited with 128: Cloning into '/atlantis-data/repos/xxx/devops/xxx-terraform-dev/12/default/aws/dev/ap-northeast-2/testnet-node/atlantis-test/.terragrunt-cache/xxx'..
fatal: could not read Username for 'https://gitlab.xxx.com': No such device or address
```

![BoltDB access denied](https://github.com/runatlantis/helm-charts/assets/103010807/0315dc8a-6a3b-4abe-8c1d-5f7e09cf33b7)

In other words, if runasuser:100 is applied, permission denied is generated in BoltDB, and if runasuser:1000 is applied, it is executed as a default user, and if the command 'atlantis plan' is sent during Merge Request, an error such as 'could not read Username' is generated.

I tried to manually solve permission-related issues by creating a Custom Dockerfile, but it didn't work out.

I would like to ask if there is a solution.

The help application version applied is as follows.
```
CHART: atlantis-4.18.0
APP VERSION: v0.26.0
```

Kubernetes Cluster version is as follows. 
`Server Version: v1.22.17-eks-8cb36c9`

Thank you for your support!

**Comment by nitrocode** [comment]
If you're customizing the image, try setting the user to Atlantis at the end

```dockerfile
USER atlantis
```

**Comment by sangkyu-korbit** [comment]
@nitrocode 

Hello!

Thank you for your advice.

However, the Dockerfile specifies 'USER: atlantis' and 'runsAsUser:100' in the values.yaml of helm chart
Setting up still causes the same error.

![error2](https://github.com/runatlantis/helm-charts/assets/103010807/1055508a-ea7a-46b8-8876-daa93a96cc9e)

**Comment by nitrocode** [comment]
What are the permissions of that file when you run `ls -la` on the file? What user owns it? Can you chown it to the atlantis user before atlantis starts and see if it works?

**Comment by sangkyu-korbit** [comment]
Hello, @nitrocode!

I'm sorry for the delay in answering.

I will describe the results of each permission setting.

**100,1000 (This is the default value for Atlantis values.yaml)**
![100,1000  result1](https://github.com/runatlantis/helm-charts/assets/103010807/44b2b98a-01df-47a5-a759-a0d737db6d9a)


![100,1000  result](https://github.com/runatlantis/helm-charts/assets/103010807/b43d87f5-3dcd-466c-a91a-9041ac2f38b7)


**1000,1000**
![1000,1000  result3](https://github.com/runatlantis/helm-charts/assets/103010807/762d6dc5-3bdf-46fd-a3c1-6cff04fc6c1a)


![1000,1000  result1](https://github.com/runatlantis/helm-charts/assets/103010807/e3770f98-195d-4afb-ae78-dacde237e931)


![1000,1000  result2](https://github.com/runatlantis/helm-charts/assets/103010807/f0022115-39c0-4fca-ac7a-90919b230aa2)

The pod runs normally, but an error occurs when 'atlantis plan' is performed in the gitlab. 
(presumed to be a problem caused by the UID being set to "1000" instead of "100)

Thank you for your help :)

**Comment by nitrocode** [comment]
Is atlantis-data a shared mount? I would login to the container as user `root` and then `cd /atlantis-data` and `ls -la` to check the permissions. It's possible that the ownership of your files is `root:root` instead of `atlantis:atlantis`.

**Comment by sangkyu-korbit** [comment]
Hello @nitrocode !

/atlantis-data is the data directory for the volume mount set in value.yam.

`Atlantis data directory: /atlantis-data`

I attach the result after setting fsGroup and runAsUser to 100 respectively.
![atlantis-data(1000)](https://github.com/runatlantis/helm-charts/assets/103010807/a815bfcc-c37e-4836-8716-6ca43d1a17a1)


![atlantis-data(100)](https://github.com/runatlantis/helm-charts/assets/103010807/6bb4db66-b898-47ba-9908-a1615e3aadf3)


Thank you for your continued help :)

**Comment by nitrocode** [comment]
Try setting the fsGroup to 1000 and if you are using alpine, set the uid to 100. If using debian, set the uid to 1000.

See https://github.com/runatlantis/atlantis/issues/4087

**Comment by sangkyu-korbit** [comment]
Hello, @nitrocode ! 

I'll try to fix it based on what I've said so far.

Thank you for your help so far :)

**Comment by sangkyu-korbit** [comment]
I'm leaving it to help those who are looking for the same phenomenon.

In my case, it operated normally after setting up IRSA.

Atlantis values.yaml chart

`  annotations:
  # IRSA example:
    eks.amazonaws.com/role-arn:`

You can set it up on the same part as above.
